test/SlowTests/Issues/RavenDB-20628.cs

```diff
@@ -40,26 +45,25 @@ public async Task ClusterWideTransactionShouldThrowTimeoutException()
             }
 
             var db = await Server.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(store.Database);
-            db.ForTestingPurposesOnly().WaitForDatabaseResultsTimeoutInClusterTransaction = (command) =>
+            db.ForTestingPurposesOnly().AfterCommitInClusterTransaction = () =>
             {
-                if (command.ParsedCommands.Any(cmd => cmd.Type == CommandType.PUT))
-                    return TimeSpan.Zero;
-                else
-                    return null;
+                return Task.Delay(15_000);
             };
 
-            var e = await Assert.ThrowsAsync<RavenException>( async () =>
+            var e = await Assert.ThrowsAsync<TaskCanceledException>( async () =>
             {
+                using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(1)))
```

Comment:
I'm afraid this might be racy and we will throw in a different place.. 
Let's do it differently... let's cancel the token after we reach `AfterCommitInClusterTransaction` and wait for the `SaveChanges` to throw.

Reply:
Done